Firmware/menu.cpp

```diff
@@ -140,6 +145,12 @@ void menu_submenu_no_reset(menu_func_t submenu, const bool feedback)
 uint8_t menu_item_ret(void)
 {
 	lcd_draw_update = 2;
+	menu_item++;
+	
+	//prevent the rest of the menu items from rendering or getting clicked
+	menu_row = LCD_HEIGHT; // early exit from the MENU_BEGIN() for loop at the end of the current cycle
+	menu_line = 0; // prevent subsequent menu items from rendering at all in the current MENU_BEGIN() for loop cycle
+	menu_clicked = 0; // prevent subsequent items from being able to be clicked in case the current menu or position was changed by the clicked menu item
```

Comment:
I don’t think we need to reset `menu_clicked`. It should be reset automatically on next `menu_begin` … `menu_end` iteration.

Reply:
I had to do this because of the following scenario:
- You click a back item in a menu
- That action changes the lcd_encoder to the parent menu value from the menu stack let's say that lcd_encoder gets set to 3 by this
- The current menu MENU_BEGIN for loop iteration still continues though, so you end up at some point to the menu_item==3==lcd_encoder, which will be clicked as well.

So to avoid a second click action in the wrong menu item if you click on an item that executes menu_goto and changes lcd_encoder, we clear the click event in menu_item_ret().

Reply:
Makes sense. Will think a bit more about this.

I would really love to see the menu code be decoupled from `lcd_encoder` in the future. `lcd_encoder` should, in my mind, only be used to track changes in the encoder position. In that way the menu code would have its own separate encoder variable in the menu stack. But this is quite a bit outside the scope of this PR.